operator/pkg/client/client_factory.go

```diff
@@ -126,6 +142,7 @@ func NewSingletonClients(config *rest.Config, logger logr.Logger) (*SingletonCli
 		openAPIParser:               openapi.NewOpenAPIParser(openAPIGetter),
 		structuredRestClientCache:   map[string]resource.RESTClient{},
 		unstructuredRestClientCache: map[string]resource.RESTClient{},
+		runtimeClient:               runtimeClient,
```

Comment:
```suggestion
		client:               client,
```
see my other comment for reasoning

Reply:
this key was removed and added inline